### PR TITLE
Fixes 10e18 bug

### PR DESCRIPTION
**Pull request by ZackAttax**
Transforms incoming and outgoing 10e18 values.

### CODE REVIEW

File: src/components/kudos/GiveKudos.jsx [hunk 1]
```diff
@@ -16,15 +16,15 @@ const sendGiveKudosInitialState = {
...
   email: '',
   name: '',
-  amount: 0,
+  amount: BigInt(0),
 }
 
-const splitU256 = (value) => {
-  const bigIntValue = BigInt(value)
-  const BigInt128 = BigInt(2) ** BigInt(128)
+const transformInt = (value) => {
```
Comment (drspacemn) [inline src/components/kudos/GiveKudos.jsx]: can we split this out into a lib folder? 
maybe `lib/kudos.js` or `lib/utils.js`?

File: src/components/kudos/GiveKudos.jsx [hunk 2]
```diff
@@ -115,16 +119,19 @@ export function GiveKudos({ userData, markStepComplete }) {
 
   useEffect(() => {
     if (givenKudosData) {
-      setKudosGiven(givenKudosData)
+      const dividedKudosData = divideBy10e18(givenKudosData)
+      setKudosGiven(dividedKudosData)
```
Comment (drspacemn) [inline src/components/kudos/GiveKudos.jsx]: thoughts on onelining this?
`setKudosGiven(divideBy10e18(givenKudosData))`?

File: src/components/kudos/GiveKudos.jsx [hunk 3]
```diff
@@ -40,11 +40,15 @@ const useKudosReadData = (args, functionName) => {
   return kudosData
 }
 
+const divideBy10e18 = (value) => {
```
Comment (drspacemn) [inline src/components/kudos/GiveKudos.jsx]: move to lib file?

File: src/components/kudos/GiveKudos.jsx (hunk 2 above)
Comment (ZackAttax) [reply src/components/kudos/GiveKudos.jsx]: I like one liners but think nested functions are ugly. We can go nested one liner here.

**Comment by ZackAttax** [comment]
Responded to comments and added guards to only allow digits and whole numbers in the amount input.

**Pull request merged**